test/functional/test-amp-img.js

```diff
@@ -374,4 +377,93 @@ describe('amp-img', () => {
     impl.unlayoutCallback();
   });
 
+  describe('blurred image placeholder', () => {
+    let el;
+    let img;
+    let impl;
+
+    beforeEach(() => {
+      toggleExperiment(iframe.win, 'blurry-placeholder', true);
+      getImgWithBlur(true, true);
+    });
+
+    afterEach(() => {
+      impl.unlayoutCallback();
+    });
+
+    function getImgWithBlur(addPlacholder, addBlurClass) {
+      el = document.createElement('amp-img');
+      img = document.createElement('img');
+      if (addPlacholder) {
+        img.setAttribute('placeholder', '');
+      }
+      if (addBlurClass) {
+        img.classList.add('i-amphtml-blur');
+      }
+      el.appendChild(img);
+      el.getResources = () => Services.resourcesForDoc(document);
+      el.getPlaceholder = () => {return img;};
+      impl = new AmpImg(el);
+    }
+
+    it('should correctly detect if a blurry image child exists', () => {
+      impl.buildCallback();
+      impl.layoutCallback();
+      expect(impl.hasBlurredPlaceHolder_).to.be.true;
+
+      getImgWithBlur(true, false);
+      impl.buildCallback();
+      impl.layoutCallback();
+      expect(impl.hasBlurredPlaceHolder_).to.be.false;
+
+      el = document.createElement('amp-img');
+      el.getPlaceholder = () => {return undefined;};
+      impl = new AmpImg(el);
+      impl.buildCallback();
+      impl.layoutCallback();
+      expect(impl.hasBlurredPlaceHolder_).to.be.false;
+
+      el = document.createElement('amp-img');
+      el.getPlaceholder = () => {return undefined;};
+      impl = new AmpImg(el);
+      impl.buildCallback();
+      impl.layoutCallback();
+      expect(impl.hasBlurredPlaceHolder_).to.be.false;
+    });
+
+    it('should fade in the image if the blurry placheolder exists', () => {
+      impl.buildCallback();
+      impl.layoutCallback();
+      expect(impl.element.classList.contains(
+          'i-amphtml-blur-loading')).to.be.true;
+
+      const loadEvent = createCustomEvent(iframe.win, 'load');
+      impl.img_.dispatchEvent(loadEvent);
+      expect(impl.element.classList.contains(
+          'i-amphtml-blur-loaded')).to.be.true;
```

Comment:
Try using `expect(el.classList).includes('i-amphtml-blur-loaded')`. This will yield a better error message on failure -- currently it'll look like "expected false to be true".

Reply:
That makes sense! As a tangentially related question, is there a stylistic preference between using 'el.classList' vs 'impl.element.classList'?

Reply:
We already have a reference to the element so just use it. :)

Reply:
ok cool! 

Also btw, when I tried the 'expect(el.classList).includes('i-amphtml-blur-loaded')', the test failed because it said the object being tested was a domtokenlist which wasn't supported by the "includes()" call. Instead, I just did 'expect(el).to.have.class('i-amphtml-blur-loaded')' which seems to give the right error. Please let me know if this is a problem!

Reply:
Even better!